substrate/frame/system/src/lib.rs

```diff
@@ -766,6 +771,20 @@ pub struct AccountInfo<Nonce, AccountData> {
 	pub data: AccountData,
 }
 
+type NonceOf<T> = <T as Config>::Nonce;
+type AccountDataOf<T> = <T as Config>::AccountData;
+type AccountInfoOf<T> = AccountInfo<NonceOf<T>, AccountDataOf<T>>;
+
+pub struct GetDefaultAccountInfo<T>(PhantomData<T>);
```

Comment:
This is a very bad name. It should be something like `SetNonceToBlockNumber`.
And make sure you provide an implementation that's matching the existing behaviour. Not everyone want's this feature.

Reply:
So you want it configurable in the pallet or something to switch it on/off?

Reply:
yes. again, not everyone wants this

Reply:
Okay we can add a `type DefaultNonce: Get<Self::Nonce>;` and default it to `GetDefault` aka. `()`.  
Although this would be a breaking change in the `frame_system` config cc @kianenigma.

Reply:
I am trying to understand why would someone opt not to use it. What would be the downside of this approach?